### DISCUSSION TITLE
How to call another function (A contract ) by (B contract) on chain

### DISCUSSION THREAD

**Discussion by yue1823** [aptos-labs/aptos-developer-discussions#271] [Questions] [answered]
### Discord user ID

._yue_.

### Describe your question in detail.

I want  to use a function by another contract `(but i don't have it original code just know how to use)`
how to call this function on my own contract on chain 
`(A contract - fun )` `(B contract - mine)`
is aptos allow directly call A-fun from B ?

# not using SDK , "from move "

### What have you tried or looked at?  Or how can we reproduce the error?

think?

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

N/A

**Comment by Cobtech** [comment]
ok

**Comment by adityaaggarwal12** [comment]
Hey Yue,

If the source code for a module is not available, you can use the contract's bytecode as a dependency to utilize its functions in your module.

To download the contract's bytecode, use the following command:
```bash
aptos move download --bytecode --account <account_address_where_contract_is_deployed> --package <package_name>
```

After downloading, add the downloaded folder as a dependency in your `Move.toml`. This will allow you to use the contract's public functions in your module.

Additionally, you can decompile the bytecode using this command:
```bash
aptos move decompile --package-path <path_to_folder_with_.mv_files>
```

For more detailed guidance, you can follow this [official guide](https://preview.aptos.dev/en/build/smart-contracts/book/packages#using-bytecode-for-dependencies:~:text=a%20Rust%20library.-,Using%20Bytecode%20for%20Dependencies,-Move%20bytecode%20can).

**Reply by yue1823** [reply]
may I ask how to add dependency to Move.toml ?

-diff (mine)
      | sources
      | Move.toml
-Liquidswap. (downaload)
      | sources
      |Move.toml
      
  I want ask what should I add dependency on move.toml 

  [dependencies]
liquidswap = { local = "../Liquidswap", addr_subst = { "liquidswap" ="0x190d } }
I put this one on diff - move.toml 
is this right ?

**Reply by adityaaggarwal12** [reply]
Yep, that looks fine!, Adding the dependency  like that should work.

**Reply by Alexzander-Stone** [reply, marked as answer]
Hey @yue1823, you can see active usage of third party dependencies [here](https://learn.aptoslabs.com/example/third-party-contract-dependency/third-party-dependencies).
Specifically, this [`Move.toml`](https://github.com/aptos-labs/move-by-examples/blob/main/multidex-router/aptos/move/Move.toml) will contain examples of sourcing from both a local file (PancakeSwap) and a git repo (Liquidswap).
